src/settings_gui.cpp

```diff
@@ -140,6 +142,52 @@ void ShowBaseSetTextfileWindow(TextfileType file_type, const TBaseSet* baseset,
 	new BaseSetTextfileWindow<TBaseSet>(file_type, baseset, content_type);
 }
 
+std::vector<int> refresh_rates = { 30, 60, 75, 90, 100, 120, 144, 240 };
+
+// We want 60Hz as the default for this pointer, though it gets overriden with whatever is in the config.
+int *current_refresh_rate = &refresh_rates[1];
+
+std::string hertz = "Hz";
+std::string current_refresh_rate_str = std::to_string(*current_refresh_rate) + hertz;
+
+// Add the refresh rate from the config and the refresh rates from all the monitors to
+// our list of refresh rates shown in the GUI.
+void addRefreshRatesAndSelect() {
+	// std::unordered_set so we can efficiently add all monitor refresh rates to this
+	// set without duplicates, as it has near constant lookup times.
+	std::unordered_set<int> tempRates(refresh_rates.begin(), refresh_rates.end());
+
+	// Add the refresh rate as selected in the config.
+	int wantedRate = _settings_client.gui.refresh_rate;
+	if (std::find(tempRates.begin(), tempRates.end(), wantedRate) == tempRates.end()) {
+		// Custom refresh rate. Add it to the list of refresh rates.
+		tempRates.insert(_settings_client.gui.refresh_rate);
```

Comment:
It is a set; why not just add it?

Reply:
`insert` is the default function to add to a `std::unordered_set`.

Reply:
sorry, you misunderstood: you first check if the entry is already in the set, then insert it. Why not just insert it, and let `std::unsorted_set` deal with if it is there or not?